db/internal_stats.h

```diff
@@ -604,6 +604,18 @@ class InternalStats {
     ++cf_stats_count_[type];
   }
 
+  void IncrNumRunningCompactionSortedRuns(uint64_t value) {
+    num_running_compaction_sorted_runs_.fetch_add(value);
+  }
+
+  void DecrNumRunningCompactionSortedRuns(uint64_t value) {
+    num_running_compaction_sorted_runs_.fetch_sub(value);
+  }
+
+  uint64_t NumRunningCompactionSortedRuns() {
+    return num_running_compaction_sorted_runs_.load();
+  }
+
```

Comment:
Just be mindful of the performance with strict sequence order - with stats, usually we don't mind the inaccuracy relaxed order brought to us. It seems to me that this can be replaced with the relaxed order since the operations within the same thread on the same variable under relaxed order should guarantee their orders so your assertions in below constructor/destructor won't fail.

Reply:
updated to use relaxed memory ordering